client/src/components/Users.jsx

```diff
@@ -0,0 +1,82 @@
+import React, { useEffect, useState } from 'react';
+import List from '@material-ui/core/List';
+import Avatar from '@material-ui/core/Avatar';
+import socketIOClient from 'socket.io-client';
+import { useHistory } from 'react-router-dom';
+import Button from '@material-ui/core/Button';
+import ListItem from '@material-ui/core/ListItem';
+import { useGetUsers } from '../services/userService';
+import { makeStyles } from '@material-ui/core/styles';
+import ListItemText from '@material-ui/core/ListItemText';
+import ListItemAvatar from '@material-ui/core/ListItemAvatar';
+import { authenticationService, useLogout } from '../services/authenticationService';
+
+const useStyles = makeStyles((theme) => ({
+	list: {
+		maxHeight: 'calc(100vh - 112px)',
+		overflowY: 'auto'
+	}
+}));
+
+const Users = (props) => {
+	const classes = useStyles();
+	const history = useHistory();
+	const [ currentUserId ] = useState(authenticationService.currentUserValue._id);
+	const [ users, setUsers ] = useState([]);
+	const [ newUser, setNewUser ] = useState(null);
+	const [ onlineUsers, setOnlineUsers ] = useState({});
+
+	const getUsers = useGetUsers();
+	const logout = useLogout();
+
+	useEffect(
+		() => {
+			getUsers().then((res) => setUsers(res));
+		},
+		[ newUser ]
+	);
+
+	useEffect(() => {
+		const socket = socketIOClient(process.env.REACT_APP_API_URL, {
+			transports: [ 'websocket', 'polling', 'flashsocket' ]
+		});
+		socket.on('onlineUsers', (data) => {
+			setOnlineUsers(data);
+		});
+		socket.on('users', (data) => {
+			setNewUser(data);
```

Comment:
Instead of triggering a whole new fetch of all new users when a new user is created, you can just pass in the socket the user details, and then append that user to the users state variable